Library/Homebrew/brew.sh

```diff
@@ -744,6 +744,30 @@ fi
 source "${HOMEBREW_LIBRARY}/Homebrew/utils/analytics.sh"
 setup-analytics
 
+# Migrate from linuxbrew-core to homebrew-core
+# HOMEBREW_CORE_MERGE_MAINTAINER can be set by the maintainers
+# who merge homebrew-core into linuxbrew-core to avoid being migrated
+# We only migrate HOMEBREW_DEVELOPER users right now
+# Don't migrate on CI either
+if [[ -n "${HOMEBREW_DEVELOPER}" ]] &&
+   [[ -z "${HOMEBREW_CORE_MERGE_MAINTAINER}" ]] &&
+   [[ -z "${GITHUB_ACTIONS}" ]] &&
+   [[ "${HOMEBREW_BASH_COMMAND}" == "${HOMEBREW_LIBRARY}/Homebrew/cmd/update.sh" ]] &&
+   [[ "https://github.com/Homebrew/linuxbrew-core" == "$(git -C "${HOMEBREW_CORE_REPOSITORY}" remote get-url origin)" ]]
+then
+  export HOMEBREW_CORE_DEFAULT_GIT_REMOTE="https://github.com/Homebrew/homebrew-core"
+  export HOMEBREW_CORE_GIT_REMOTE="${HOMEBREW_CORE_DEFAULT_GIT_REMOTE}"
+
+  if [[ "$(git config homebrew.linuxbrewmigrationcomplete 2>/dev/null)" != "true" ]]
+  then
+    export HOMEBREW_LINUXBREW_MIGRATION=1
+  fi
+
+  git -C "${HOMEBREW_CORE_REPOSITORY}" remote set-url origin "${HOMEBREW_CORE_DEFAULT_GIT_REMOTE}"
+  git -C "${HOMEBREW_CORE_REPOSITORY}" fetch origin master
+  git -C "${HOMEBREW_CORE_REPOSITORY}" reset --hard origin/master
```

Comment:
`brew update` can do this for you so there's no need to do this here: https://github.com/Homebrew/brew/blob/b117a1888eec84e1e33cf220b63d04371ef0bb3f/Library/Homebrew/cmd/update.sh#L61-L69

Reply:
@iMichka Sorry, I linked to the wrong place, the relevant code is
https://github.com/Homebrew/brew/blob/b117a1888eec84e1e33cf220b63d04371ef0bb3f/Library/Homebrew/cmd/update.sh#L473-L481

That's what we should be relying on instead of duplicating here. It may be worth introducing a function to unify both of these.